src/components/ComboBox/ComboBox.jsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Tag } from 'carbon-components-react';
+
+import { settings } from '../../constants/Settings';
+
+import CarbonComboBox from './CarbonComboBox';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  // eslint-disable-next-line react/forbid-foreign-prop-types
+  ...CarbonComboBox.propTypes,
+  loading: PropTypes.bool,
+  // Callback that is called with the value of the input on change
+  onChange: PropTypes.func.isRequired,
+  // OPtional classname to be applied to wrapper
+  wrapperClassName: PropTypes.string,
+  // String to pass for tags close button aria-label. Will be prepended to value name
+  closeButtonText: PropTypes.string,
+  // Bit that will allow mult value and tag feature
+  hasMultiValue: PropTypes.bool,
+};
+
+const defaultProps = {
+  ...CarbonComboBox.defaultProps,
+  loading: false,
+  inline: false,
+  wrapperClassName: null,
+  closeButtonText: 'Close',
+  hasMultiValue: false,
+  items: [],
+};
+
+const ComboBox = ({
+  inline,
+  loading,
+  wrapperClassName,
+  closeButtonText,
+  hasMultiValue,
+  itemToString,
+  onChange,
+  items,
+  ...comboProps
+}) => {
+  // Ref for the combobox input
+  const comboRef = React.createRef();
+  const { downshiftProps } = comboProps;
+  // Current selected item that shows in the input
+  const [selectedItem, setSelectedItem] = useState(null);
+  // Array that populates list
+  const [listItems, setListItems] = useState(items);
+  // Array that populates tags
+  const [tagItems, setTagItems] = useState([]);
+
+  useEffect(() => {}, []);
+
+  useEffect(() => {
+    // If there are tags then clear and focus the input
+    if (tagItems.length >= 1) {
+      setSelectedItem('');
+      comboRef.current.textInput.current.focus();
+    }
+  });
+
+  const handleOnKeypress = evt => {
+    // Current value of input
+    const currentValue = comboRef.current.textInput.current.value.trim();
+
+    // Only apply logic if user hit enter
+    if (evt.key === 'Enter' && currentValue !== '' && currentValue !== undefined) {
+      // Check if there is already a tag for this value and return it
+      const filteredTags = tagItems.filter(x => itemToString(x) === currentValue);
+      // Check if value is part of items array
+      const filteredItems = listItems.filter(x => itemToString(x) === currentValue);
```

Comment:
same here, there is a lot of `filteredItems.length < 1` and `filteredItems[0]`, we could use a constant like `matchedItem` that either is undefined or is the item